Valigator.Tests/DataEqualityTests.cs

```diff
@@ -0,0 +1,199 @@
+﻿using FluentAssertions;
+using Functional;
+using System;
+using System.Linq;
+using Xunit;
+
+namespace Valigator.Tests
+{
+	public class DataEqualityTests
+	{
+		[Fact]
+		public void RecordEqualsItself()
+		{
+			var guid = Guid.NewGuid();
+			var uniqueGuids = new[] { Guid.NewGuid(), Guid.NewGuid(), Guid.NewGuid() };
+			var item1 = TestRecord.CreateTestRecord(guid, uniqueGuids);
+			item1.Equals(item1).Should().BeTrue();
+			item1.ID.Equals(item1.ID).Should().BeTrue();
+			item1.UniqueGuid.Equals(item1.UniqueGuid).Should().BeTrue();
+			item1.ArrayOfUniqueGuid.Equals(item1.ArrayOfUniqueGuid).Should().BeTrue();
+			item1.UniqueName.Equals(item1.UniqueName).Should().BeTrue();
+		}
+
+		[Fact]
+		public void RecordsEqual()
+		{
+			var guid = Guid.NewGuid();
+			var uniqueGuids = new[] { Guid.NewGuid(), Guid.NewGuid(), Guid.NewGuid() };
+
+			var item1 = TestRecord.CreateTestRecord(guid, uniqueGuids);
+			var item2 = TestRecord.CreateTestRecord(guid, uniqueGuids);
+
+			item1.Equals(item2).Should().BeTrue();
+			(item1 == item2).Should().BeTrue();
+			item1.Should().Be(item2);
+
+		}
+
+		[Fact]
+		public void RecordsDoNotEqual()
+		{
+			var guid = Guid.NewGuid();
+			var uniqueGuids = new[] { Guid.NewGuid(), Guid.NewGuid(), Guid.NewGuid() };
+
+			var item1 = TestRecord.CreateTestRecord(guid, uniqueGuids);
+			var item2 = TestRecord.CreateTestRecord(Guid.NewGuid(), uniqueGuids);
+
+			(item1.Equals(item2)).Should().BeFalse();
+			(item1 == item2).Should().BeFalse();
+		}
+
+		[Fact]
+		public void RemoveDescriptors()
+		{
+			var guid = Guid.NewGuid();
+			var uniqueGuids = new[] { Guid.NewGuid(), Guid.NewGuid(), Guid.NewGuid() };
+
+			var item1 = TestRecord.CreateTestRecord(guid, uniqueGuids);
+
+			item1.UniqueName.Equals(item1.UniqueNameWithDifferentDataDescriptors).Should().BeFalse();
+		}
+
+		[Fact]
+		public void CheckHashCode()
+		{
+			var guid = Guid.NewGuid();
+			var uniqueGuids = new[] { Guid.NewGuid(), Guid.NewGuid(), Guid.NewGuid() };
+
+			var item1 = TestRecord.CreateTestRecord(guid, uniqueGuids);
+
+			item1.ArrayOfUniqueGuid.GetHashCode();
```

Comment:
No assertion on this test. Presumably this should create an item2 and assert that the HashCodes from both are the same.

Reply:
Although it looks like you have a test later on that does that, what is the point in this test?

Reply:
This was just to check functionality as I was building.
I've now removed it.
CompareRecordHashCode should cover what you're describing